test/Microsoft.Health.Fhir.Shared.Tests.Integration/Persistence/FhirStorageTests.cs

```diff
@@ -721,6 +732,41 @@ public async Task GivenADeletedResource_WhenUpdatingSearchParameterIndexAsync_Th
             }
         }
 
+        [Fact]
+        [FhirStorageTestsFixtureArgumentSets(DataStore.CosmosDb)]
+        public async Task GivenAValidResource_WhenUpdatingAResourceWithSameDataImmaterialKeepHistoryValue_ServerShouldNotCreateANewVersionAndReturnOk()
```

Comment:
I feel this test would work better as a theory with the keepHistory value as the input argument. That way the two are tested individually and the duplicate code isn't needed.